gnunet-gtk fail to build

**Describe the bug**
```
[davidak@gaming:~]$ nix run nixpkgs.gnunet-gtk -c gnunet-gtk
builder for '/nix/store/w1a305jwbbnn2h723w0hwl1xlflhg2w9-gnunet-gtk-0.13.1.drv' failed with exit code 1; last 10 log lines:
  checking gnutls/abstract.h presence... yes
  checking for gnutls/abstract.h... yes
  checking for gnutls_pubkey_import in -lgnutls... yes
  checking for GNUnet core... --with-gnunet not specified
  checking for gnunetutil >= 0.9.0... yes
  checking GNUNET_CFLAGS... -I/nix/store/aawda90f2iv5x0dxp3ag74i37bclwc80-gnunet-0.14.0/include
  checking GNUNET_LIBS... -L/nix/store/aawda90f2iv5x0dxp3ag74i37bclwc80-gnunet-0.14.0/lib -lgnunetutil
  checking for gnunet/gnunet_conversation_service.h... no
  checking for gnunet/gnunet_fs_service.h... no
  configure: error: compiling gnunet-gtk requires GNUnet core headers
[0 built (1 failed), 6 copied (33.3 MiB), 9.2 MiB DL]
error: build of '/nix/store/w1a305jwbbnn2h723w0hwl1xlflhg2w9-gnunet-gtk-0.13.1.drv' failed
```

**To Reproduce**
Steps to reproduce the behavior:
1. ...
2. ...
3. ...

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
@pstn and @jonringer edited it last

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.1, NixOS, 21.05pre273435.0aeba64fb26 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"home-manager-20.09, nixos-21.05pre273435.0aeba64fb26, nixos-hardware, nixos-unstable-21.05pre273435.0aeba64fb26"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`


Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/networking/p2p/gnunet/default.nix
+++ b/pkgs/applications/networking/p2p/gnunet/default.nix
@@ -5,11 +5,11 @@
 
 stdenv.mkDerivation rec {
   pname = "gnunet";
-  version = "0.14.0";
+  version = "0.14.1";
 
   src = fetchurl {
     url = "mirror://gnu/gnunet/${pname}-${version}.tar.gz";
-    sha256 = "sha256-2u9gO9Mu0dM1yixcaqOnZcDfGcp69dc5CH5tkl6vRas=";
+    sha256 = "1hhqv994akymf4s593mc1wpsjy6hccd0zbdim3qmc1y3f32hacja";
   };
 
   enableParallelBuilding = true;

--- a/pkgs/applications/networking/p2p/gnunet/gtk.nix
+++ b/pkgs/applications/networking/p2p/gnunet/gtk.nix
@@ -5,6 +5,7 @@
 , gtk3
 , libextractor
 , libgcrypt
+, libsodium
 , libxml2
 , pkg-config
 , wrapGAppsHook
@@ -12,11 +13,11 @@
 
 stdenv.mkDerivation rec {
   pname = "gnunet-gtk";
-  version = "0.13.1";
+  version = "0.14.0";
 
   src = fetchurl {
     url = "mirror://gnu/gnunet/${pname}-${version}.tar.gz";
-    sha256 = "1zdzgq16h77w6ybwg3lqjsjr965np6iqvncqvkbj07glqd4wss0j";
+    sha256 = "18rc7mb45y17d5nrlpf2p4ixp7ir67gcgjf4hlj4r95ic5zi54wa";
   };
 
   nativeBuildInputs= [
@@ -31,15 +32,16 @@ stdenv.mkDerivation rec {
     gtk3
     libextractor
     libgcrypt
+    libsodium
     libxml2
   ];
+
+  configureFlags = [ "--with-gnunet=${gnunet}" ];
 
   patchPhase = "patchShebangs pixmaps/icon-theme-installer";
 
   meta = gnunet.meta // {
     description = "GNUnet GTK User Interface";
     homepage = "https://git.gnunet.org/gnunet-gtk.git";
-    # configure: error: compiling gnunet-gtk requires GNUnet core headers
-    broken = true;
   };
 }

